web/src/debug.ts

```diff
@@ -10,42 +12,36 @@
     evaluates to foo() and prints the elapsed time
     to the console along with the name "foo". */
 
-export function print_elapsed_time(name, fun) {
+export function print_elapsed_time<T>(name: string, fun: () => T): T {
     const t0 = Date.now();
     const out = fun();
     const t1 = Date.now();
-    console.log(name + ": " + (t1 - t0) + " ms");
+    console.log(`${name} :  ${t1 - t0} ms`);
     return out;
 }
 
-export function check_duplicate_ids() {
-    const ids = new Set();
-    const collisions = [];
+export function check_duplicate_ids(): {collisions: Collision[]; total_collisions: number} {
+    const ids = new Set<string>();
+    const collisions: Collision[] = [];
     let total_collisions = 0;
 
-    for (const o of Array.prototype.slice.call(document.querySelectorAll("*"))) {
+    const element: NodeList = document.querySelectorAll("*");
```

Comment:
`element` is not a good name for this variable. Why did you extract it to a variable at all?

Reply:
The TLS was complaining and it was not inferring the right types so it was neccesary to extract the element and annotate it with the correct types

Reply:
It had no effect.

The underlying problem is that TypeScript thinks `o` is of type `any` because it doesn’t understand `Array.prototype.slice.call`. We don’t need the `Array.prototype.slice.call` anyway; it’s for ancient browsers where `NodeList` isn’t [directly iterable](https://caniuse.com/mdn-api_domtokenlist_--iterator). Let’s remove it.